### ISSUE TITLE
Pythonic filtering and transformation

### ISSUE THREAD

**Issue by jakevdp** [altair-viz/altair#208]
Currently the `Transform` object takes strings of javascript to filter & transform data. We could create a more Pythonic API that would generate and validate the appropriate javascript for most use-cases; for example, instead of

``` python
Chart(data).transform_data(calculate=[Formula('b2', expr='2 * datum.b')],
                           filter="datum.b2 > 60")
```

We could instead have something like

``` python
d = Calculator(data)
Chart(data).transform_data(Calculate(b2=2 * d.b),
                           filter=d.b > 60)
```

Or something similar. Any thoughts?

**Comment by TomAugspurger** [comment]
There's also the "magic X" used in https://github.com/kieferk/dfply/blob/master/README.md#the-pandas-ply-x-dataframe-symbol https://github.com/kieferk/dfply/blob/master/README.md, where `X.b` would resolve to the original `data` passed to chart.

**Comment by jakevdp** [comment]
Thanks! I ended up providing something quite similar to this, though I called it ``df`` for lack of a better name.

I think using ``X`` in the context of altair would be confusing, because we already have ``X`` as a channel name.

If you have a minute to look at #215, I'd love to hear your thoughts on how I ended up implementing this. You can read an overview in ``transform.rst`` there, though you'll have to look at the raw restructured text at the moment.

**Comment by TomAugspurger** [comment]
Neat, will review tonight!

**Comment by jakevdp** [comment]
Closed by #215
